### ISSUE TITLE
Is it the same to pass properties to JDBC URL, ClickHouseDataSource and DriverManager.getConnection?

### ISSUE THREAD

**Issue by kell18** [yandex/clickhouse-jdbc#245]
Currently there are 3 ways to pass properties to JDBC driver:
1. JDBC URL get arguments, like: `jdbc:clickhouse://localhost:8123/default?p1=1&p2=2`
2. ClickHouseDataSource args: `ClickHouseDataSource(url, properties)`
3. Generic DriverManager args: `DriverManager.getConnection(url, props)`

Are this tree the same?

**Comment by zhicwu** [comment]
Yes they should be the same. Added [this test case](https://github.com/ClickHouse/clickhouse-jdbc/blob/e59bd468c1e1b049d84a0a8183758740735e8e9a/clickhouse-jdbc/src/test/java/com/clickhouse/jdbc/ClickHouseDataSourceTest.java#L16) for regression.
